[ADD] feat(BitstreamLicense) Bitstream's license default value when nothing is selected by the user

Pourbaix opened the pull request:
First commit for consumers, others will follow...

If a user creates a bitstream for a submission, it will have a default license of CC-BY.

#### NOTE:
The config file `uclouvain.cfg` should be re-organized because it is starting to become messy.

Commit:
[ADD] feat(BitstreamLicense) Bitstream's license default value when nothing is selected by the user

If a user creates a bitstream for a submission, it will have a default license of CC-BY.

Authored-By: Michaël Pourbaix <michael.pourbaix@uclouvain.be>

zannkukai commented on dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/consumer/LicenseConsumer.java:
Maybe it will be better to simplify our folders structures : not creating a "submissionMetadataGenerators" but create a main "consumers" folder and put any uclouvain consumers into it.

zannkukai commented on dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/consumer/LicenseConsumer.java:
```diff
@@ -0,0 +1,46 @@
+    @Override
+    public void consume(Context context, Event event) throws Exception {
+        // Only process bitstreams that have no license && are being modified
+        if (event.getSubjectType() == Constants.BITSTREAM && (event.getEventType() == Event.MODIFY_METADATA || event.getEventType() == Event.MODIFY)) {
```
COSMETIC : Could be 
```java
Arrays.asList([Event.MODIFY, Event.MODIFY_METADATA]).contains(event.getEventType())
```

zannkukai commented on dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/consumer/LicenseConsumer.java:
```diff
@@ -0,0 +1,46 @@
+        // Only process bitstreams that have no license && are being modified
+        if (event.getSubjectType() == Constants.BITSTREAM && (event.getEventType() == Event.MODIFY_METADATA || event.getEventType() == Event.MODIFY)) {
+            Bitstream bitstream = (Bitstream) event.getSubject(context);
+            if (bitstreamService.getMetadataFirstValue(bitstream, "dc", "rights", "license", null) == null) {
```
Here you can use `getMetadataFirstValue(T, MetadataFieldName, language)` method signature.
The `MetadataFieldName` could be a bean or could be create into the class constructor based on property from configuration. 
I used the second option in my PR.... because i'm not yet fluent with bean ; but if multiple classes use same logic, it could make sense.

```java
MetadataFieldName accessMetadataFieldName;
...
constructorMethod() {
    String fieldName = configurationService.getProperty("global.license.metadata.field", "dc.rights.license");
    this.accessMetadataFieldName = new MetadataFieldName(fieldName);
}
```

zannkukai commented on dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/consumer/LicenseConsumer.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class LicenseConsumer implements Consumer {
+
+    String LICENSE_URL = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("uclouvain.bitstream.default.license", "https://creativecommons.org/licenses/by/4.0/");
```
UPPER_CASE variable name should be reserve for public static variable (java best practice)

zannkukai commented on dspace/config/dspace.cfg:
```diff
@@ -802,7 +802,7 @@ event.dispatcher.default.class = org.dspace.event.BasicDispatcher
 # Add iiif here, if you are using dspace-iiif.
 # Add orcidqueue here, if the integration with ORCID is configured and wish to enable the synchronization queue functionality
-event.dispatcher.default.consumers = versioning, discovery, eperson, dedup, crisconsumer, orcidqueue, audit, nbeventsdelete, referenceresolver, orcidwebhook, itemenhancer, customurl, reciprocal, filetypemetadataenhancer
+event.dispatcher.default.consumers = versioning, discovery, eperson, dedup, crisconsumer, orcidqueue, audit, nbeventsdelete, referenceresolver, orcidwebhook, itemenhancer, customurl, reciprocal, filetypemetadataenhancer, licensegenerator
```
into default `dspace.cfg` or (better ?) redefined into `local.cfg` ?

zannkukai commented on dspace/config/dspace.cfg:
```diff
@@ -814,6 +814,9 @@ related-item-enhancer-poller.delay = 5000
 event.dispatcher.noindex.class = org.dspace.event.BasicDispatcher
 event.dispatcher.noindex.consumers = eperson
 
+event.consumer.licensegenerator.class = org.dspace.uclouvain.submissionMetadataGenerators.consumer.LicenseConsumer
+event.consumer.licensegenerator.filters = Bitstream
```
same

Commit:
[FIX] feat(pdf-attestation): Download button not appearing for manager when validating.

Fixed a bug where the manager could not see the download attestation button when validating an item.
This was due to the fact that we could not retrieve the owning collection of the item when in workflow validation.

@Authored-By: Pourbaix Michaël <michael.pourbaix@uclouvain.be>
